pages/views/elements.py

```diff
@@ -246,19 +246,36 @@ def get_context_data(self, **kwargs):
         user_enumerated_progress = EnumeratedProgress.objects.filter(
             progress=sequence_progress)
 
-        has_completed_sequence = True
-        for element in enumerated_elements:
-            user_element_progress = user_enumerated_progress.filter(rank=element.rank).first()
-            if not user_element_progress or user_element_progress.viewing_duration < element.min_viewing_duration:
-                has_completed_sequence = False
-                break
+        last_element = enumerated_elements.last().page_element
+        is_last_element_certificate = hasattr(last_element, 'certificate')
+
+        if is_last_element_certificate:
+            # Exclude last element from the check, if it is a certificate
+            enumerated_elements = enumerated_elements.exclude(pk=enumerated_elements.last().pk)
+
+        completed_elements_subquery = user_enumerated_progress.filter(
```

Comment:
Really cool subquery using `OuterRef`. Does that change the minimum Django version required to run djaodjin-pages?